infra/testing/sw-env.js

```diff
@@ -11,18 +11,34 @@
  limitations under the License.
  */
 
-const mockFetch = require('./mock-fetch');
+const serviceWorkerMock = require('service-worker-mock');
 const {IDBFactory, IDBKeyRange} = require('shelving-mock-indexeddb');
-const makeServiceWorkerEnv = require('service-worker-mock');
+const Blob = require('./sw-env-mocks/Blob');
+const Event = require('./sw-env-mocks/Event');
+const {addEventListener, dispatchEvent} = require('./sw-env-mocks/event-listeners');
+const ExtendableEvent = require('./sw-env-mocks/ExtendableEvent');
+const fetch = require('./sw-env-mocks/fetch');
+const Headers = require('./sw-env-mocks/Headers');
+const Request = require('./sw-env-mocks/Request');
+const SyncEvent = require('./sw-env-mocks/SyncEvent');
+const SyncManager = require('./sw-env-mocks/SyncManager');
 
-global.fetch = mockFetch;
+// Assign all properties of `self` to `global`;
+Object.assign(global, serviceWorkerMock());
+global.self = global;
```

Comment:
I'm still confused at why this self is needed. serviceWorkerMock creates a self and this will overwrite that.

Reply:
It's not necessarily needed, but if we don't do this then all the code below would need to be like this:

```js
global.Blob = global.self.Blob = Blob;
global.Event = global.self.Event = Event;
// ...
```

So I guess it's a matter of which we prefer. I think what I have is fine since it's actually the same way browsers implement it, i.e. self (and window) is a property of the global object that points to the global object. For example: `window.window === window`, `window.self === window`, `self.self === self`, etc.